### PR TITLE
Merge all changes from SHAP Community Fork

### LINKED ISSUE #1081: keras is no longer supported, please use tf.keras instead.
while running example given on homepage of SHAP-  "Deep learning example with DeepExplainer (TensorFlow/Keras models) " getting this error. Please suggest. ( is there no support for keras only and we have to use tensorflow's keras)?

### LINKED ISSUE #2144: Categorical features in LGBM cause failure when using __call__ method of the TreeExplainer
I bumped into the problem that when using the `__call__` method of the `TreeExplainer` on a LGBM model that uses categorical features, one gets an error like _"ValueError: could not convert string to float:"_.

The origin seem to be [this line](https://github.com/slundberg/shap/blob/89855e03d7e92ba0d2ec463c0505878f718664e3/shap/explainers/_tree.py#L206), i.e. one converts the input DataFrame into a numpy array and thus the info about categorical fetures (when those are defined by the types of features) get's lost, when the data are passed to the `shap_values` method [here](https://github.com/slundberg/shap/blob/89855e03d7e92ba0d2ec463c0505878f718664e3/shap/explainers/_tree.py#L211). The `shap_values` method itself works fine when the complete DataFrame is passed as input. 

Below is a minimal reproducible example:
```python
import lightgbm as lgb
import shap

# train XGBoost model
X, y = shap.datasets.adult()
# add a dummy categorical variable
X["Categorical"] = "A"
X["Categorical"] = X["Categorical"].astype("category")
# train a LGBM model with a categorical feature in the data
model = lgb.LGBMClassifier().fit(X, y)

# compute SHAP values
explainer = shap.TreeExplainer(model)
# this works
_ = explainer.shap_values(X.head(100))
# this fails
_ = explainer(X.head(100))
```

My setup:
python 3.8
lightgbm 3.2.1
shap 0.39.0

### LINKED ISSUE #2471: Error for shap.plots.waterfall when show is False
Hi 

I've got the following error when I set 'show' to False calling shap.plots.waterfall. It was fine when show was True.
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
<ipython-input-77-215661f2f39e> in <module>
----> 1 gcf1 = shap.plots.waterfall(shap_values_test[0], max_display=50, show=False)

~\AppData\Local\Continuum\anaconda3\envs\grenoble\lib\site-packages\shap\plots\_waterfall.py in waterfall(shap_values, max_display, show)
     41     # Turn off interactive plot
     42     if show is False:
---> 43         plt.ioff()
     44 
     45 

NameError: name 'plt' is not defined

Another question, how to save the result figure into a file, .png or .jpg? Thank you very much.

### LINKED ISSUE #2528: Future Warning from sklearn for KernelExplainer concerning LassoLarsIC
Calling the method `shap_values` from the  `KernelExplainer` gives the following warning:
```
The default of 'normalize' will be set to False in version 1.2 and deprecated in version 1.4.

If you wish to scale the data, use Pipeline with a StandardScaler in a preprocessing stage. To reproduce the previous behavior:

from sklearn.pipeline import make_pipeline

model = make_pipeline(StandardScaler(with_mean=False), LassoLarsIC())

If you wish to pass a sample_weight parameter, you need to pass it as a fit parameter to each step of the pipeline as follows:

kwargs = {s[0] + '__sample_weight': sample_weight for s in model.steps}
model.fit(X, y, **kwargs)

Set parameter alpha to: original_alpha * np.sqrt(n_samples). 
```
Versions:
python: 3.9.12 (conda)
sklearn: 1.0.1
shap: 0.40.0
OS: Ubuntu 18.04.1 LTS

### LINKED ISSUE #2668: shap_values deprecation warning throws error
I am trying to run a DART booster using '0.41.0' and in using shap_values() I get an error 

warnings.warn("shap_values() is deprecated; use __call__().", warnings.DeprecationWarning)

AttributeError: module 'warnings' has no attribute 'DeprecationWarning'
Snag_1ac4d05

### LINKED ISSUE #2706: DeprecationWarning when Importing display from IPython.core.display
Turning on `filterwarnings = error` in pytest errors when importing shap with ipython >= 8:

```
.tox/py39/lib/python3.9/site-packages/shap/__init__.py:38: in <module>
    from .plots._beeswarm import summary_legacy as summary_plot
.tox/py39/lib/python3.9/site-packages/shap/plots/__init__.py:6: in <module>
    from ._force import force, initjs
.tox/py39/lib/python3.9/site-packages/shap/plots/_force.py:11: in <module>
    from IPython.core.display import display, HTML
.tox/py39/lib/python3.9/site-packages/IPython/core/display.py:44: in __getattr__
    warn(f"Importing {name} from IPython.core.display is deprecated since IPython 7.14, please import from IPython display", DeprecationWarning, stacklevel=2)
E   DeprecationWarning: Importing display from IPython.core.display is deprecated since IPython 7.14, please import from IPython display
```

https://github.com/ipython/ipython/commit/43a01a21101ea65dbd52cf760c1e4f149bfab588

### LINKED ISSUE #2738: Shouldn't the data sampling method use `replace=True` ?
I was looking at the `Tabular` class (in shap/shap/maskers/_tabular.py) and saw that the input data is sampled using the `utils.sample` function (defined in shap/shap/utils/_general.py).

This function samples a dataset using `sklearn.utils.resample(X, n_samples=nsamples, random_state=random_state)`. The default behavior of the sklearn's function is to implement resampling **with replacement**.
Is this wanted here ?

```python
import pandas as pd
import numpy as np
from shap.utils._general import sample

# Fake data
df = pd.DataFrame({'x': [i for i in range(100)], 'y': [i / 10 for i in range(100)]})

# Sample 99 data (will use replacement)
sampled_data_99 = sample(df.values, 99)
len(np.unique(sampled_data_99 , axis=0))  # -> 62

# Sample 100 data (will take all the dataset)
sampled_data_100 = sample(df.values, 100)
len(np.unique(sampled_data_100 , axis=0))  # -> 100
```

### LINKED ISSUE #2771: warnings.warn("shap_values() is deprecated; use __call__().", warnings.DeprecationWarning)
While running explainer on my dataframe getting the below error.
explainer = shap.Explainer(model_predict, score_df)
shap_values = explainer.shap_values(score_df)

---> 36 shap_values = explainer.shap_values(score_df)
     
      

/local_disk0/.ephemeral_nfs/cluster_libraries/python/lib/python3.9/site-packages/shap/explainers/_permutation.py in shap_values(self, X, npermutations, main_effects, error_bounds, batch_evals, silent)
    212             of such matrices, one for each output.
    213         """
--> 214         warnings.warn("shap_values() is deprecated; use __call__().", warnings.DeprecationWarning)
    215 
    216         explanation = self(X, max_evals=npermutations * X.shape[1], main_effects=main_effects)

AttributeError: module 'warnings' has no attribute 'DeprecationWarning'

### LINKED ISSUE #2831: Can't install using python 3.11 or with numpy >= 1.24 on python < 3.11
I noticed this issue originally when trying to install mlflow on python 3.11 which depends on shap (see https://github.com/mlflow/mlflow/issues/7681). The reason that shap fails to install is that it depends on numba which in turn requires llvmlite which does not work with python 3.11 yet.
Numba 0.56.4 also requires numpy < 1.24 which is required for python 3.11. 
Not sure if numba is a hard requirement for shap or if it can be made optional to fix this. Otherwise this ticket could track the issue.

### LINKED ISSUE #2837: numpy dtypes have to be corrected for numpy version 1.24
Hi,
since [numpy 1.24](https://numpy.org/doc/stable/release/1.24.0-notes.html) some `dtype` aliases have been removed. Therefore, in the code `np.bool` and `np.int` have to be replaced by `np.bool_` and `np.int_`, resp.
Regards, Jörg.

### LINKED ISSUE #2866: Test failures with scikit-learn > 1.2
`load_boston` was deprecated in 1.0 and was removed in 1.2 by `scikit-learn`. 

```bash
============================= test session starts ==============================
platform linux -- Python 3.10.10, pytest-7.2.0, pluggy-1.0.0
Matplotlib: 3.6.2
Freetype: 2.12.1
rootdir: /build/source, configfile: pytest.ini, testpaths: tests
plugins: mpl-0.16.1
collected 118 items / 1 deselected / 117 selected                              

tests/explainers/test_kernel.py .....xxx...                              [  9%]
tests/explainers/test_linear.py .....xxxxx...xxxxx                       [ 24%]
tests/explainers/test_permutation.py .sssssssss                          [ 33%]
tests/explainers/test_sampling.py ....                                   [ 36%]
tests/explainers/test_tree.py sxsssssss.x.xsxxx..x.xxxxxx.xxxx.ss.ssssss [ 72%]
sss                                                                      [ 75%]
tests/maskers/test_custom.py F                                           [ 76%]
tests/maskers/test_image.py ....                                         [ 79%]
tests/maskers/test_tabular.py FFFF                                       [ 82%]
tests/plots/test_bar.py s                                                [ 83%]
tests/plots/test_decision.py .                                           [ 84%]
tests/plots/test_dependence.py ..                                        [ 86%]
tests/plots/test_dependence_string_features.py .....                     [ 90%]
tests/plots/test_force.py FF                                             [ 92%]
tests/plots/test_heatmap.py ss                                           [ 94%]
tests/plots/test_image.py ...                                            [ 96%]
tests/plots/test_text.py .                                               [ 97%]
tests/plots/test_violin.py s                                             [ 98%]
tests/plots/test_waterfall.py ss                                         [100%]

=================================== FAILURES ===================================
______________________________ test_raw_function _______________________________

    def test_raw_function():
        """ Make sure passing a simple masking function works.
        """
    
>       X, _ = shap.datasets.boston()

tests/maskers/test_custom.py:12: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/nix/store/63gixr2mi1jryln1yb1vbyvb833rmyyy-python3.10-shap-0.41.0/lib/python3.10/site-packages/shap/datasets.py:34: in boston
    d = sklearn.datasets.load_boston()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

name = 'load_boston'

    def __getattr__(name):
        if name == "load_boston":
            msg = textwrap.dedent(
                """
                `load_boston` has been removed from scikit-learn since version 1.2.
    
                The Boston housing prices dataset has an ethical problem: as
                investigated in [1], the authors of this dataset engineered a
                non-invertible variable "B" assuming that racial self-segregation had a
[...]
```

`load_bosten` is used at [two places](https://github.com/slundberg/shap/search?q=load_boston).

### LINKED ISSUE #2878: DeprecationWarnings in shap/explainers/_tree.py
Firstly, thank you very much for the awesome package! 

We are using the TreeExplainer and Plotting utilities. Since we update some of our dependency (matplolib, numpy, xgboost) the shap library produces some DeprecationWarings (see the cleaned logs attached). 

Our setup: 
* python 3.10
* xgboost  1.7.2
* matplotlib 3.6.2
* numpy 1.22.4

```
/shap/explainers/_tree.py:1511: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    self.leaf_child_cnt.append(np.zeros(self.num_nodes[i], dtype=np.int))

/shap/explainers/_tree.py:1520: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    self.children_default = np.zeros(shape, dtype=np.int)

 /shap/explainers/_tree.py:1521: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    self.features = np.zeros(shape, dtype=np.int)

 /shap/explainers/_tree.py:1375: DeprecationWarning: NPY_ARRAY_UPDATEIFCOPY, NPY_ARRAY_INOUT_ARRAY, and NPY_ARRAY_INOUT_FARRAY are deprecated, use NPY_WRITEBACKIFCOPY, NPY_ARRAY_INOUT_ARRAY2, or NPY_ARRAY_INOUT_FARRAY2 respectively instead, and call PyArray_ResolveWritebackIfCopy before the array is deallocated, i.e. before the last call to Py_DECREF.
    self.max_depth = _cext.compute_expectations(

/shap/explainers/_tree.py:1375: DeprecationWarning: UPDATEIFCOPY detected in array_dealloc.  Required call to PyArray_ResolveWritebackIfCopy or PyArray_DiscardWritebackIfCopy is missing
    self.max_depth = _cext.compute_expectations(

/xgboost/core.py:122: UserWarning: ntree_limit is deprecated, use `iteration_range` or model slicing instead.
    warnings.warn(

shap/plots/_beeswarm.py:340: UserWarning: No data for colormapping provided via 'c'. Parameters 'vmin', 'vmax' will be ignored
    pl.scatter(shaps[nan_mask], pos + ys[nan_mask], color="#777777", vmin=vmin,

/shap/plots/_beeswarm.py:365: MatplotlibDeprecationWarning: Unable to determine Axes to steal space for Colorbar. Using gca(), but will raise in the future. Either provide the *cax* argument to use as the Axes for the Colorbar, provide the *ax* argument to steal space from it, or add *mappable* to an Axes.
    cb = pl.colorbar(m, ticks=[0, 1], aspect=80)
```

Thank you very much.

### LINKED ISSUE #2904: RuntimeError: Cannot install on Python version 3.11.2; only versions >=3.7,<3.11 are supported.

### LINKED ISSUE #2909: NumbaDeprecationWarning with latest dependencies, as "nopython" argument is not supplied to numba.jit
When SHAP is imported with the latest version of `numba`, shap raises many NumbaDeprecationWarnings:

- Python 3.11
- Numba 0.57.0
- SHAP 0.41.0

> The 'nopython' keyword argument was not supplied to the 'numba.jit' decorator. The implicit default value for this argument is currently False, but it will be changed to True in Numba 0.59.0. See https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit for details.

```
/shap/utils/_clustering.py:34: NumbaDeprecationWarning: The 'nopython' keyword argument was not supplied to the 'numba.jit' decorator. The implicit default value for this argument is currently False, but it will be changed to True in Numba 0.59.0. See https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit for details.
  @jit
/shap/utils/_clustering.py:53: NumbaDeprecationWarning: The 'nopython' keyword argument was not supplied to the 'numba.jit' decorator. The implicit default value for this argument is currently False, but it will be changed to True in Numba 0.59.0. See https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit for details.
  @jit
/shap/utils/_clustering.py:62: NumbaDeprecationWarning: The 'nopython' keyword argument was not supplied to the 'numba.jit' decorator. The implicit default value for this argument is currently False, but it will be changed to True in Numba 0.59.0. See https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit for details.
  @jit
/shap/utils/_clustering.py:68: NumbaDeprecationWarning: The 'nopython' keyword argument was not supplied to the 'numba.jit' decorator. The implicit default value for this argument is currently False, but it will be changed to True in Numba 0.59.0. See https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit for details.
  @jit
/shap/utils/_clustering.py:76: NumbaDeprecationWarning: The 'nopython' keyword argument was not supplied to the 'numba.jit' decorator. The implicit default value for this argument is currently False, but it will be changed to True in Numba 0.59.0. See https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit for details.
  @jit
/shap/links.py:4: NumbaDeprecationWarning: The 'nopython' keyword argument was not supplied to the 'numba.jit' decorator. The implicit default value for this argument is currently False, but it will be changed to True in Numba 0.59.0. See https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit for details.
  @numba.jit
/shap/links.py:9: NumbaDeprecationWarning: The 'nopython' keyword argument was not supplied to the 'numba.jit' decorator. The implicit default value for this argument is currently False, but it will be changed to True in Numba 0.59.0. See https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit for details.
  @numba.jit
/shap/links.py:14: NumbaDeprecationWarning: The 'nopython' keyword argument was not supplied to the 'numba.jit' decorator. The implicit default value for this argument is currently False, but it will be changed to True in Numba 0.59.0. See https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit for details.
  @numba.jit
/shap/links.py:19: NumbaDeprecationWarning: The 'nopython' keyword argument was not supplied to the 'numba.jit' decorator. The implicit default value for this argument is currently False, but it will be changed to True in Numba 0.59.0. See https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit for details.
  @numba.jit
/shap/utils/_masked_model.py:362: NumbaDeprecationWarning: The 'nopython' keyword argument was not supplied to the 'numba.jit' decorator. The implicit default value for this argument is currently False, but it will be changed to True in Numba 0.59.0. See https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit for details.
  @jit # we can't use this when using a custom link function...
/shap/utils/_masked_model.py:384: NumbaDeprecationWarning: The 'nopython' keyword argument was not supplied to the 'numba.jit' decorator. The implicit default value for this argument is currently False, but it will be changed to True in Numba 0.59.0. See https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit for details.
  @jit
/shap/utils/_masked_model.py:427: NumbaDeprecationWarning: The 'nopython' keyword argument was not supplied to the 'numba.jit' decorator. The implicit default value for this argument is currently False, but it will be changed to True in Numba 0.59.0. See https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit for details.
  @jit
/shap/utils/_masked_model.py:438: NumbaDeprecationWarning: The 'nopython' keyword argument was not supplied to the 'numba.jit' decorator. The implicit default value for this argument is currently False, but it will be changed to True in Numba 0.59.0. See https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit for details.
  @jit
/shap/maskers/_tabular.py:185: NumbaDeprecationWarning: The 'nopython' keyword argument was not supplied to the 'numba.jit' decorator. The implicit default value for this argument is currently False, but it will be changed to True in Numba 0.59.0. See https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit for details.
  @jit
/shap/maskers/_tabular.py:196: NumbaDeprecationWarning: The 'nopython' keyword argument was not supplied to the 'numba.jit' decorator. The implicit default value for this argument is currently False, but it will be changed to True in Numba 0.59.0. See https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit for details.
  @jit
/shap/maskers/_image.py:174: NumbaDeprecationWarning: The 'nopython' keyword argument was not supplied to the 'numba.jit' decorator. The implicit default value for this argument is currently False, but it will be changed to True in Numba 0.59.0. See https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit for details.
  @jit
/shap/explainers/_partition.py:675: NumbaDeprecationWarning: The 'nopython' keyword argument was not supplied to the 'numba.jit' decorator. The implicit default value for this argument is currently False, but it will be changed to True in Numba 0.59.0. See https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit for details.
```

### LINKED ISSUE #2911: Shap not compatible with numpy>=1.24.0
Since 1.24.0 numpy removed `np.bool` which will lead to various errors using `shap` repository.
On my case I fixed `numpy` version to 1.23.0 to make it work.

### LINKED ISSUE #2944: Meta-issue: merge SHAP Community Fork
This is a meta-issue to track progress towards merging the [SHAP Community Fork](https://github.com/dsgibbons/shap) into this upstream project.

See discussion: https://github.com/slundberg/shap/discussions/2942

TODO:

- [x] Fix failing CodeQL checks: https://github.com/dsgibbons/shap/pull/104
- [x] Prepare branch that removes fork-specific changes: https://github.com/dsgibbons/shap/pull/103
- [x] Complete main outstanding PRs in dsgibbons/shap:
   - [x] https://github.com/dsgibbons/shap/pull/102
   - [x] https://github.com/dsgibbons/shap/pull/101
   - [x] https://github.com/dsgibbons/shap/pull/99
   - [x] https://github.com/dsgibbons/shap/pull/97
   - [x] https://github.com/dsgibbons/shap/pull/105
- [x] PR to merge from fork into upstream: https://github.com/slundberg/shap/pull/2943
  - [x] In the description, link all relevant issues in upstream repository that will be closed
      - Done a first-pass, but there may be more
  - [x] Reviews from SHAP Community Fork maintainers
  - [x] Merge without squashing

### PR DESCRIPTION
**Pull request by connortann**
Meta-PR to merge all changes from the dsgibbons [SHAP Community Fork](https://github.com/dsgibbons/shap).
Pending ongoing discussion: https://github.com/slundberg/shap/discussions/2942

Meta-issue: resolves #2944 .

## Related issues that will be fixed

Issues that are fixed on the community fork, and will be closed by this PR: [list is sorted numerically]

- Closes #1081 
- Closes #2144
- Related to #2322 
- Closes #2471 
- Closes #2528 
- Closes #2668 
- Closes #2706 
- Closes #2738
- Closes #2771
- Closes #2831
- Closes #2837 
- Closes #2866 
- Closes #2878 
- Closes #2904 
- Closes #2909
- Closes #2911

## Duplicated PRs

PRs here at `slundberg/shap` that have already been covered by the commits over at `dsgibbons/shap`, thus can be closed once this PR is merged. [list is sorted numerically]

- Duplicates #1802
- Duplicates #1890 
- Duplicates #1931 
- Duplicates #2020 
- Duplicates #2048 
- Duplicates #2166 
- Duplicates #2595
- Duplicates #2641
- Duplicates #2649 
- Duplicates #2655
- Duplicates #2672 
- Duplicates #2679
- Duplicates #2726 
- Duplicates #2769
- Duplicates #2781
- Duplicates #2792
- Duplicates #2830 
- Duplicates #2885 
- Duplicates #2890 
- Duplicates #2912 
- Duplicates #2924 
- Duplicates #2932
- Duplicates #2939

Plus many further fixes, as per the closed issues and PRs on the Community Fork.

### CODE REVIEW

File: CHANGELOG.md
Comment (dsgibbons) [inline CHANGELOG.md]: Should we keep this?

File: docs/index.rst
Comment (dsgibbons) [inline docs/index.rst]: This can be removed

File: CHANGELOG.md
Comment (connortann) [reply CHANGELOG.md]: I think we probably should keep this. I'll push a commit to reword the last section here. In future, we could include it in the readthedocs documentation - but suggest parking that improvement for a furture PR.

**Review by dsgibbons — APPROVED** [review]
Looks great! Thank you for putting this together so quickly.

**Comment by thatlittleboy** [comment]
I'm looking through the changes now, please give me a sec~

**Comment by connortann** [comment]
I will rebase again on top of `dsgibbons/master`, to include the most recent PR https://github.com/dsgibbons/shap/pull/100

File: CHANGELOG.md
Comment (dsgibbons) [inline CHANGELOG.md]: The tagged issue and PR numbers are for dsgibbons/shap. Do we need to add an identifier to reference the fork explicitly?

Comment (connortann) [reply CHANGELOG.md]: Good suggestion, I'll do that now. We will probably want to overhaul the changelog in near future, e.g. host as part of the docs.

Comment (thatlittleboy) [reply CHANGELOG.md]: yep, I was just about to comment this as well. I'll tag along here:

---

What do we think about all of these PR references in the CHANGELOG? They're no longer correct once we merge it into `slundberg/shap`. Shall we prepend the repo name `dsgibbons#xx` to all of these? And also this PR number?

How about something like

```
- Removed the unused `mimic.py` file and `MimicExplainer` code (dsgibbons#53, #2943).
```
?

This ensures everyone can locate the original PR where the discussions happened (even if/after we migrate to the organization)

Comment (dsgibbons) [reply CHANGELOG.md]: > How about something like
> ...

Looks good!

**Review by thatlittleboy — COMMENTED** [review]
Everything else looks good to me. I'll add on one more comment,

We've lost pre-commit-CI here. It's _not a blocker_ for this PR, but I imagine we would want to introduce it back sooner rather than later. I see 3 options:

1. Ask @slundberg to introduce pre-commit-ci on our behalf. (cc @dsgibbons )
2. Migrate over to the organization so we can set it up ourselves.
3. Barring those, we can run pre-commit as a GH action just as an interim measure. Set up in a separate PR, not here.

---

<s>ps. And also perhaps hold off for just a while longer before the merge, let's see if we can squeeze in dsgibbons#97 PR in here as well.</s>

File: CHANGELOG.md
Comment (connortann) [reply CHANGELOG.md]: I tried adding full URLs, how does that look to you? (I was most of the way there when I saw your comment.)

```markdown
- Deprecated the Boston house price dataset
  ([#38](https://github.com/dsgibbons/shap/pull/38)).
- Removed the unused `mimic.py` file and `MimicExplainer` code
  ([#53](https://github.com/dsgibbons/shap/pull/53)).
```
  
Renders as:
  
- Deprecated the Boston house price dataset ([#38](https://github.com/dsgibbons/shap/pull/38)).
- Removed the unused `mimic.py` file and `MimicExplainer` code ([#53](https://github.com/dsgibbons/shap/pull/53)).
  
FWIW, we will need to tweak the changelog more in the near future, so I suggest we park any further adjustments to a future PR.

File: tests/utils/test_general.py
Comment (thatlittleboy) [inline tests/utils/test_general.py]: here

File: .github/workflows/run_tests.yml
Comment (thatlittleboy) [inline .github/workflows/run_tests.yml]: here

File: .github/workflows/run_tests.yml
Comment (thatlittleboy) [inline .github/workflows/run_tests.yml]: here

File: tests/explainers/test_tree.py
Comment (thatlittleboy) [inline tests/explainers/test_tree.py]: here

File: tests/explainers/test_tree.py
Comment (thatlittleboy) [inline tests/explainers/test_tree.py]: here

**Review by thatlittleboy — CHANGES_REQUESTED** [review]
My bad, suddenly realized that we put in plenty of issue/PR references in-code as well.

I don't think these need the URL treatment, just a simple `dsgibbons#xx` should do the trick.

File: CHANGELOG.md
Comment (thatlittleboy) [reply CHANGELOG.md]: Works for me!

**Comment by connortann** [comment]
> My bad, suddenly realized that we put in plenty of issue/PR references in-code as well.

Here's a separate PR to implement those docs changes: https://github.com/dsgibbons/shap/pull/105 . As this PR is already quite big enough!

EDIT: Done, rebased and force pushed.

File: CHANGELOG.md
Comment (connortann) [reply CHANGELOG.md]: Any further comments on this over at https://github.com/dsgibbons/shap/pull/105

**Review by thatlittleboy — APPROVED** [review]
cat-wearing-shades-letsgo

**Review by dsgibbons — APPROVED** [review]

**Comment by ricardobarroslourenco** [comment]
@connortann do you know when those will become available as a release?

**Comment by connortann** [comment]
@ricardobarroslourenco we are working on broadening the pool of maintainers, which in turn will hopefully enable us to make a release. I don't have a fixed timeline, but I would hope over the next few weeks. See https://github.com/slundberg/shap/discussions/2942 for more context

**Pull request merged**
